commit c69d3c0246b72461f205c0d201aaf7c3ca98117d
Author: Chao Peng <chao.p.peng@linux.intel.com>
Date:   Tue Apr 26 02:27:00 2016 -0400

    i386: remove find_i440fx
    
    We already have generic acpi_get_i386_pci_host() so there is no reason to
    keep spefific find_i440fx().
    
    Signed-off-by: Chao Peng <chao.p.peng@linux.intel.com>

diff --git a/hw/i386/acpi-build.c b/hw/i386/acpi-build.c
index 6477003..2181122 100644
--- a/hw/i386/acpi-build.c
+++ b/hw/i386/acpi-build.c
@@ -435,12 +435,16 @@ static void *acpi_set_bsel(PCIBus *bus, void *opaque)
 
 static void acpi_set_pci_info(void)
 {
-    PCIBus *bus = find_i440fx(); /* TODO: Q35 support */
+    PCIBus *bus;
     unsigned bsel_alloc = 0;
+    Object *pci_host = acpi_get_i386_pci_host();
 
-    if (bus) {
-        /* Scan all PCI buses. Set property to enable acpi based hotplug. */
-        pci_for_each_bus_depth_first(bus, acpi_set_bsel, NULL, &bsel_alloc);
+    if (pci_host) {
+        bus = PCI_HOST_BRIDGE(pci_host)->bus;
+        if (bus) {
+            /* Scan all PCI buses. Set property to enable acpi based hotplug. */
+            pci_for_each_bus_depth_first(bus, acpi_set_bsel, NULL, &bsel_alloc);
+        }
     }
 }
 
diff --git a/hw/pci-host/piix.c b/hw/pci-host/piix.c
index df2b0e2..6be439a 100644
--- a/hw/pci-host/piix.c
+++ b/hw/pci-host/piix.c
@@ -412,14 +412,6 @@ PCIBus *i440fx_init(const char *host_type, const char *pci_type,
     return b;
 }
 
-PCIBus *find_i440fx(void)
-{
-    PCIHostState *s = OBJECT_CHECK(PCIHostState,
-                                   object_resolve_path("/machine/i440fx", NULL),
-                                   TYPE_PCI_HOST_BRIDGE);
-    return s ? s->bus : NULL;
-}
-
 /* PIIX3 PCI to ISA bridge */
 static void piix3_set_irq_pic(PIIX3State *piix3, int pic_irq)
 {
diff --git a/include/hw/i386/pc.h b/include/hw/i386/pc.h
index 2666619..7f7aa0c 100644
--- a/include/hw/i386/pc.h
+++ b/include/hw/i386/pc.h
@@ -306,7 +306,6 @@ PCIBus *i440fx_init(const char *host_type, const char *pci_type,
                     MemoryRegion *pci_memory,
                     MemoryRegion *ram_memory);
 
-PCIBus *find_i440fx(void);
 /* piix4.c */
 extern PCIDevice *piix4_dev;
 int piix4_init(PCIBus *bus, ISABus **isa_bus, int devfn);
